scipy/sparse/coo.py

```diff
@@ -137,16 +137,13 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                 except TypeError:
                     raise TypeError('invalid input format')
 
-                self.row = np.array(ij[0], copy=copy)
-                self.col = np.array(ij[1], copy=copy)
-                self.data = np.array(obj, copy=copy)
-
+                row, col = ij
```

Comment:
Are you sure this is safe? I would have moved the `dtype=` stuff here. Now if a list is passed in, we get a slow Python `max` call in the validation code below.